catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/SearchFormsLoader.java

```diff
@@ -100,32 +102,39 @@ public static boolean enabled() {
    * <p>Caution should be exercised when executing code as system. Results from querying as system
    * are never returned to the client or cached. This means there is no risk of data leak.
    *
-   * @param framework the catalog framework, for creating system templates.
+   * @param catalogFramework the catalog framework, for creating system templates.
    * @param util for querying the catalog.
    * @param systemTemplates system templates loaded from config.
    */
   public static void bootstrap(
-      CatalogFramework framework, EndpointUtil util, List<Metacard> systemTemplates) {
-    Set<String> queryTitles =
-        executeAsSystem(() -> titlesTransform(util.getMetacardsByFilter(QUERY_TEMPLATE_TAG)));
-    Set<String> resultTitles =
-        executeAsSystem(() -> titlesTransform(util.getMetacardsByFilter(ATTRIBUTE_GROUP_TAG)));
+      CatalogFramework catalogFramework, EndpointUtil util, List<Metacard> systemTemplates) {
+    Subject systemSubject = SECURITY.runAsAdmin(SECURITY::getSystemSubject);
+    if (systemSubject == null) {
+      throw new SecurityException("Could not get systemSubject to initialize system templates");
+    }
 
-    List<Metacard> dedupedTemplateMetacards =
-        Stream.concat(
-                systemTemplates
-                    .stream()
-                    .filter(QueryTemplateMetacard::isQueryTemplateMetacard)
-                    .filter(metacard -> !queryTitles.contains(metacard.getTitle())),
-                systemTemplates
-                    .stream()
-                    .filter(AttributeGroupMetacard::isAttributeGroupMetacard)
-                    .filter(metacard -> !resultTitles.contains(metacard.getTitle())))
-            .collect(Collectors.toList());
+    ScheduledThreadPoolExecutor executor = new ScheduledThreadPoolExecutor(4);
 
-    if (!dedupedTemplateMetacards.isEmpty()) {
-      executeAsSystem(() -> framework.create(new CreateRequestImpl(dedupedTemplateMetacards)));
-    }
+    Failsafe.with(
+            new RetryPolicy()
+                .retryOn(Collections.singletonList(Exception.class))
+                .withMaxRetries(5)
+                .withBackoff(2, 60, TimeUnit.SECONDS))
+        .with(executor)
+        .onSuccess(
+            connection -> {
+              LOGGER.info("Successfully loaded System Search Forms", connection);
+            })
+        .onFailure(
+            failure -> {
+              LOGGER.error("Failed to load System Search Forms", failure);
```

Comment:
❓ Could we add on failure/success (or if there's someway to do a `finally` with failsafe) that shuts down the executor since we should be done with it?

Reply:
`onFailure` is finally he could do that here I believe.

Reply:
`onComplete` happens for both success and failure.

https://github.com/jhalterman/failsafe#event-listeners

Reply:
done